src/SprykerEco/Zed/Unzer/Business/Credentials/UnzerCredentialsValidator/UnzerCredentialsUniqueMerchantReferenceValidator.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Business\Credentials\UnzerCredentialsValidator;
+
+use Generated\Shared\Transfer\MessageTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsConditionsTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsCriteriaTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsResponseTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsTransfer;
+use SprykerEco\Shared\Unzer\UnzerConstants;
+use SprykerEco\Zed\Unzer\Business\Reader\UnzerReader;
+
+class UnzerCredentialsUniqueMerchantReferenceValidator implements UnzerCredentialsValidatorInterface
+{
+    /**
+     * @var string
+     */
+    protected const ERROR_MESSAGE_MERCHANT_REFERENCE_ALREADY_EXIST = 'Provided merchant reference already exists!';
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Reader\UnzerReader|\SprykerEco\Zed\Unzer\Business\Credentials\UnzerCredentialsValidator\UnzerToMerchantFacadeInterface
+     */
+    protected $unzerReader;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\Business\Reader\UnzerReader $unzerReader
+     */
+    public function __construct(UnzerReader $unzerReader)
+    {
+        $this->unzerReader = $unzerReader;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\UnzerCredentialsTransfer $unzerCredentialsTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerCredentialsResponseTransfer
+     */
+    public function validate(UnzerCredentialsTransfer $unzerCredentialsTransfer): UnzerCredentialsResponseTransfer
+    {
+        $unzerCredentialsResponseTransfer = (new UnzerCredentialsResponseTransfer())->setIsSuccessful(true);
+        if (
+            !in_array((int)$unzerCredentialsTransfer->getTypeOrFail(), UnzerConstants::UNZER_CHILD_CONFIG_TYPES, true)
+            || $unzerCredentialsTransfer->getMerchantReference() === null
+        ) {
```

Comment:
You can provide separate methods for some huge/unclear checks to improve readability.

Reply:
unfortunately conditions are not re-used so it will a one-line wrapper method which is used only in one place. Also method name will not explain both conditions